src/test/unit/types/ExceptionTest.cpp

```diff
@@ -45,4 +46,13 @@ TEST(ExceptionTest, Basic) {
     ASSERT_EQ(std::string(e.getStackTrace()), std::string("TB"));
   }
 }
+
+TEST(ExceptionTest, Argument_Streaming)
+  try {
+  NTA_THROW << "This msg";
+} catch (const Exception &e) {
+  EXPECT_STREQ(e.getMessage(), "This msg");
+  EXPECT_STREQ(e.what(), "Exception: ExceptionTest.cpp(52) message: This msg");
```

Comment:
The order will make no difference because `e.getMessage( )` was already called once in the copy constructor so message_ already contains the streamed data as explained above.

Oh, and I did try both ways to confirm.

Reply:
I just discovered a situation where the copy would not occur.
If the Exception object in the try block ends up on the stack in the same place in the catch block then the copy does not occur.  

This happens in the other two test cases which I had not tried until this morning.  If the NTA_THROW macro is used, it is inside an 'if' block so its stack location changed thus it forces the copy, but not in the other two test cases.

So, you are right, I still need the call to getMessage( ) when creating what_.

You already merged so I will include that change in my next PR which is about ready to be pushed.